lib/counter.js

```diff
@@ -108,8 +108,7 @@ class Counter extends Metric {
 
 	async get() {
 		if (this.collect) {
-			const v = this.collect();
-			if (v instanceof Promise) await v;
+			const v = await Promise.resolve(this.collect());
```

Comment:
The original code was intended to avoid entering the microtask queue unless necessary, for performance (i.e. don't `await` unless necessary).

What about duck-typing for a [thenable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenables)?

Reply:
That's an interesting point I didn't consider. I suppose duck-typing is fair. Updated the PR.